gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/resources/handler/authorization/AuthorizationRequestFailureHandler.java

```diff
@@ -289,4 +290,13 @@ private void doRedirect(RoutingContext context, String url) {
         cleanSession(context);
         context.response().putHeader(HttpHeaders.LOCATION, url).setStatusCode(302).end();
     }
+
+    /**
+     * Application, only with 'client_credentials' grant type cannot have
+     * `redirect_uri` in the request
+     */
+    private boolean invalidParamRedirectURI(Client client, String redirectURI) {
+        return client != null && client.getAuthorizedGrantTypes().contains(CLIENT_CREDENTIALS) &&
```

Comment:
**question**: Don't think it is useful but do we have to test also if `client.getAuthorizedGrantTypes` is null or not ?

Reply:
done, null check added